src/main/java/com/pusher/client/channel/impl/PrivateEncryptedChannelImpl.java

```diff
@@ -0,0 +1,128 @@
+        super(channelName, factory);
+        this.connection = connection;
+        this.authorizer = authorizer;
+    }
+
+    /* Base class overrides */
+
+    @Override
+    public void bind(final String eventName, final SubscriptionEventListener listener) {
+
+        if (!(listener instanceof PrivateEncryptedChannelEventListener)) {
+            throw new IllegalArgumentException(
+                    "Only instances of PrivateEncryptedChannelEventListener can be bound " +
+                            "to a private encrypted channel");
+        }
+
+        super.bind(eventName, listener);
+    }
+
+    protected byte[] checkAuthentication() {
+
+        try {
+            final Map authResponseMap = GSON.fromJson(getAuthResponse(), Map.class);
+            final String authKey = (String) authResponseMap.get("auth");
+            final String sharedSecret = (String) authResponseMap.get("shared_secret");
+
+            if (authKey == null || sharedSecret == null) {
+                throw new AuthorizationFailureException("Didn't receive all the fields expected " +
+                        "from the Authorizer, expected an auth token and shared_secret.");
+            } else {
+                secretBoxOpener = new SecretBoxOpener(Base64.decode(sharedSecret));
+                return authKey.getBytes();
+            }
+
+        } catch (final AuthorizationFailureException e) {
+            throw e; // pass this upwards
+        } catch (final Exception e) {
+            // any other errors need to be captured properly and passed upwards
+            throw new AuthorizationFailureException("Unable to parse response from Authorizer", e);
+        }
+    }
+
+    @Override
+    public String toSubscribeMessage() {
+
+        byte[] authKey = checkAuthentication();
+
+        // create the data part
+        final Map<Object, Object> dataMap = new LinkedHashMap<Object, Object>();
+        dataMap.put("channel", name);
+        dataMap.put("auth", new String(authKey));
+
+        // create the wrapper part
+        final Map<Object, Object> jsonObject = new LinkedHashMap<Object, Object>();
+        jsonObject.put("event", "pusher:subscribe");
+        jsonObject.put("data", dataMap);
+
+        return GSON.toJson(jsonObject);
+    }
+
+    @Override
+    public void updateState(ChannelState state) {
+        super.updateState(state);
+
+        if (state == ChannelState.UNSUBSCRIBED) {
+            tearDownChannel();
+        }
+    }
+
+    private void tearDownChannel() {
+        if (secretBoxOpener != null) {
+            secretBoxOpener.clearKey();
+            secretBoxOpener = null;
+        }
+    }
+
+    private String getAuthResponse() {
+        final String socketId = connection.getSocketId();
+        return authorizer.authorize(getName(), socketId);
+    }
+
+    @Override
+    protected String[] getDisallowedNameExpressions() {
+        return new String[] { "^(?!private-encrypted).*" };
```

Comment:
```suggestion
        return new String[] { "^(?!private-encrypted-).*" };
```

I think the final `-` is a compulsory part of the prefix

Reply:
👍 i thought I'd got them all 😅